src/Integration.Vsix/SonarLintTagger/README.md

```diff
@@ -82,3 +82,47 @@ Error list integration:
 - `SinkManager` gets updates through factories.
 
 
+
+Tracking document edits:
+------------------------
+See MS docs "Inside the editor": https://docs.microsoft.com/en-us/visualstudio/extensibility/inside-the-editor?view=vs-2019
+
+A text file opened in the VS Editor is represented as an ITextBuffer. The buffer can be thought of as series of
+immutable ITextSnapshots.  A region of text in a ITextSnapshot is represented by a SnapshotSpan.
```

Comment:
So SnapshotSpan is also immutable, right? (since the whole ITextSnapshot is thrown away when the file is edited)

Reply:
I'm not sure if the span is immutable but it is tied to a specific snapshot. The snapshot isn't thrown away when the buffer is edited. It's still valid, and represents the state of the document at a point in time (it will presumably be garbage collected if it's not being referenced).

A span represents a range of characters in the buffer. There can be multiple spans associated with a snapshot.